src/include/mpir_objects.h

```diff
@@ -274,18 +274,17 @@ typedef int Handle_ref_count;
 
 #elif MPICH_THREAD_REFCOUNT == MPICH_REFCOUNT__LOCKFREE
 
-#include "opa_primitives.h"
-typedef OPA_int_t Handle_ref_count;
+typedef MPL_atomic_int_t Handle_ref_count;
 
 #define MPIR_Object_set_ref(objptr_,val)                        \
     do {                                                        \
-        OPA_store_int(&(objptr_)->ref_count, val);              \
+        MPL_atomic_store_int(&(objptr_)->ref_count, val); \
```

Comment:
Let's split this to another PR and just focus on the pool changes.

Reply:
OK.